api/apps/api/test/utils/api-application.ts

```diff
@@ -30,6 +32,8 @@ export const bootstrapApplication = async (
     .useValue(fakeProjectChecker)
     .overrideProvider(FileRepository)
     .useClass(TempStorageRepository)
+    .overrideProvider(ScenarioCalibrationRepo)
+    .useClass(FakeScenarioCalibrationRepo)
```

Comment:
using a fake test double here helps us to work around the variability of time required to run an actual calibration, however, aren't we ending up confirming every time we run this test suite that the data we set in the fake repo is indeed correct in shape?

I would suggest to either accept the liability of using the real repo, wait for the calibration to finish and so on, or at least extend the result generation backwards to the point where we read data from a Marxan output file, in which case we'd at least test that the way we parse the output files and generate result payloads from there is valid and doesn't break in the future.

Probably the "using the full calibration workflow" option may be the easiest here, but I don't have strong opinions. If doing so, I'd say a long-enough sleep could be ok as a first step, but in general we may want to implement a proper await via polling of status, similarly to what we do in bots or the frontend does, but this would be definitely overkill for the moment, even though it's only a few dozens lines of code.

Reply:
We tried to implement the test in the way that you point out (waiting until the blm calibration process finishes) but we realized that bullmq queues implementation is mocked, meaning that geoprocessing and api services cannot communicate between them 🤔

Reply:
Since blm calibration process starts after adding a job to the corresponding queue, we didn't manage to run the calibration in the e2e test

Reply:
Maybe we could 'invoke' in some way geoprocessing service in order to manually start the calibration process. I'll try it :+1:

Reply:
ah yes, you are right.

actually - looking at the relevant code more carefully now (sorry, I should have done that earlier), I think that given that what we send in the response is pretty much what we get out of the repository, I would say that maybe all we need is to keep your current test implementation, and just add explicit typing to the fake data we use for the fake repository, what do you think? I'll suggest the relevant change inline separately.

I feel that although we could plug back into the calibration process, this would require more effort than useful for the value that we're trying to achieve here.

Reply:
It makes sense :+1: